front/lib/api/assistant/generation.ts

```diff
@@ -185,6 +185,12 @@ export async function renderConversationForModel({
     if (tokensUsed + c <= allowedTokenCount) {
       tokensUsed += c;
       selected.unshift(messages[i]);
+    } else if (messages[i].role === "content_fragment") {
```

Comment:
Aren't we supposed to truncate differently based on the content fragment type? slack vs injected by users?

Reply:
That's a good question. Do you think we should truncate Slack differently? I can see arguments both ways so I don't have a strong opinion about it.

Reply:
For sure slack should favor the most recent part of the thread.

Reply:
Ok so I need to bring the content fragment object down here into the code then. I will look into it.

Reply:
Oh it's there already. 👍